mms/protocol/otf_message_handler.py

```diff
@@ -0,0 +1,241 @@
+            for idx, val in enumerate(ret):
+                msg += struct.pack("!i", len(req_id_map[idx]))
+                msg += struct.pack('!{}s'.format(len(req_id_map[idx])), req_id_map[idx].encode())
+
+                # msg += struct.pack('!i', 200) # CODE
+
+                if isinstance(val, str):
+                    content_type = "text"
+                    msg += struct.pack('!i', len(content_type))
+                    msg += struct.pack('!{}s'.format(len(content_type)), content_type.encode())
+
+                    msg += struct.pack('!i', len(val))
+                    msg += struct.pack('!{}s'.format(len(val)), val.encode())
+                elif isinstance(val, bytes):
+                    content_type = "binary"
+                    msg += struct.pack('!i', len(content_type))
+                    msg += struct.pack('!{}s'.format(len(content_type)), content_type.encode())
+
+                    msg += struct.pack('!i', len(val))
+                    msg += val
+                else:
+                    content_type = 'json'
+                    json_value = json.dumps(val)
+                    msg += struct.pack('!i', len(content_type))
+                    msg += struct.pack('!{}s'.format(len(content_type)), content_type.encode())
+
+                    msg += struct.pack('!i', len(json_value))
+                    msg += struct.pack('!{}s'.format(len(json_value)), json_value.encode())
+
+            for req in invalid_reqs.keys():
+                msg += struct.pack('!i', len(req))
+                msg += struct.pack('!{}s'.format(len(req)), req.encode())
+
+                msg += struct.pack('!i', 4)
+                msg += struct.pack('!i', invalid_reqs.get(req))
+                msg += struct.pack('!i', len("Invalid input provided".encode()))
+                msg += struct.pack('!{}s'.format(len("Invalid input provided".encode())),
+                                   "Invalid input provided".encode())
+                content_type = "text"
+                msg += struct.pack('!i', len(content_type))
+                msg += struct.pack('!{}s'.format(len(content_type)), content_type.encode())
+            msg += struct.pack('!i', -2)  # End of list
+
+            return msg
+
+        except Exception:
+            # TODO: Return a invalid response
+            raise MMSError(Err.ENCODE_FAILED, "Invalid message received")
+
+    def _encode_response(self, kwargs):
+        msg = bytearray()
+        try:
+            msg += struct.pack('!i', int(kwargs['code']))
+            msg_len = len(kwargs['message'])
+            msg += struct.pack('!i', msg_len)
+            msg += struct.pack('!{}s'.format(msg_len), kwargs['message'].encode())
+            if 'predictions' in kwargs and kwargs['predictions'] is not None:
+                msg += kwargs['predictions']
+            else:
+                msg += struct.pack('!i', 0)  # no predictions
+            msg += bytes("\r\n".encode())
+        except Exception:
```

Comment:
No exception is expected

Reply:
Catching this to log a proper message in case of codec failures.